compiler/rustc_middle/src/mir/visit.rs

```diff
@@ -376,14 +394,46 @@ macro_rules! make_mir_visitor {
                 }
             }
 
+            fn super_statement_debuginfos(
+                &mut self,
+                stmt_debuginfo: & $($mutability)? [StmtDebugInfo<'tcx>],
+                location: Location
+            ) {
+                for debuginfo in stmt_debuginfo {
+                    self.visit_statement_debuginfo(debuginfo, location);
+                }
+            }
+
+            fn super_statement_debuginfo(
+                &mut self,
+                stmt_debuginfo: & $($mutability)? StmtDebugInfo<'tcx>,
+                location: Location
+            ) {
+                match stmt_debuginfo {
+                    StmtDebugInfo::AssignRef(local, place) => {
+                        self.visit_local(
+                            $(& $mutability)? *local,
+                            PlaceContext::NonUse(NonUseContext::VarDebugInfo),
+                            location
+                        );
+                        self.visit_place(
+                            place,
+                            PlaceContext::NonUse(NonUseContext::VarDebugInfo),
```

Comment:
We can now have `visit_place` called with `NonUse`.
But `super_place` keeps transforming those into a `NonMutatingUse`. This also means we can have `ProjectionElem::Index` called with `NonUse`.
Should we adapt `super_place` and `super_projection_elem` to keep them `NonUse`?